image_writers_guide/guidelines.adoc

```diff
@@ -5,143 +5,186 @@
+
+
+*Cleaning Temporary Files*
+
+Every temporary files you create during the build process should be removed. This includes any files added with the `ADD` command, too.  For example, it is strongly recommended you run the `yum clean` command after performing `yum install` operations.
+
+If you craft your `RUN` statement as follows, you can prevent the `yum` cache from ending up in an image layer:
+
+----
+RUN yum -y install mypackage && yum -y install myotherpackage && yum clean all -y
+----
 
 Be aware that if you instead write:
-		
-		RUN yum -y install mypackage
-		RUN yum -y install myotherpackage && yum clean all -y
 
-Then the first `yum` invocation will leave extra files in that layer which cannot be removed by the later clean operation.  (The extra files will not be visible in the final image but they will be present in the underlying layers).
+----
+RUN yum -y install mypackage
+RUN yum -y install myotherpackage && yum clean all -y
+----
+
+Then the first `yum` invocation leaves extra files in that layer which cannot be removed by the later `yum clean` operation. The extra files are not visible in the final image, but they are present in the underlying layers.
+
+The current Docker build process does not allow a command run in a later later to shrink the space used by the image when something was removed in an earlier layer. However, this may change in the future. This means that if you perform an `rm` command in a later layer, the files are hidden but it does not reduce the overall size of the image that needs to be downloaded. Therefore, as with the `yum clean` example, it is best to remove files in the same command that created them, where possible, so they do not end up written to a layer.
+
+In addition, performing multiple commands in a single `RUN` statement reduces the number of layers in your image, which improves download and extraction time.
+
+*Placing Instructions in Proper Order*
 
-Although the current Docker build process does not allow shrinking the space used by the image when we remove something from an earlier layer by a command run in a later layer, this may change in the future.   This means that if you perform an `rm` command in a later layer, you will hide the files but you will not reduce the overall size of the image that needs to be downloaded.  Therefore, as with the yum clean example, it's best to remove files in the same command that created them, where possible, so they do not end up written to a layer.  
+Docker reads the [sysitem]#Dockerfile# and runs the instructions from top to bottom. Every successfully executed instruction creates a layer which can be reused the next time this or another image is built. It is very important to place instructions that will rarely change at the top of your [sysitem]#Dockerfile#. Doing so ensures the next builds of the same image are very fast because the cache is not invalidated by upper layer changes.
```

Comment:
[Suggestion]
Every successfully executed instruction creates a layer which can be reused the next time this or another image is built. ==>> Every instruction that is executed successfully creates .....